File: solutions/Plus-One/Plus-One.py


class Solution:
    # @param digits, a list of integer digits
    # @return a list of integer digits
    def plusOne(self, digits):
        ans=[]
        flag=1
        for i in range(len(digits)-1,-1,-1):
            temp=digits[i]+flag
            if temp>9:
                temp-=10
                flag=1
            else:
                flag=0
            ans.append(temp)
        if flag:
            ans.append(flag)
        ans.reverse()
        return ans
 

File: solutions/Plus-One/Plus-One.cpp

class Solution {
public:
    vector<int> plusOne(vector<int> &digits) {
        for(int i=digits.size()-1;i>=0;i--)
        {
            digits[i]++;
            if(digits[i]<10)
                break;
            digits[i]=0;
        }
        if(digits[0]==0)
        {
            digits[0]=1;
            digits.push_back(0);
        }
        return digits;
    }
};


 